### ISSUE TITLE
Option to specify input format

### ISSUE THREAD

**Issue by jaykaron** [mapbox/tippecanoe#606]
Hello,

As noted in the documentation Tippecanoe assumes GeoJSON format from standard input.
Is there a way to specify the input format?
I am trying to pipe CSV data. When saved as a CSV file everything works as expected, but get a parsing error when piping it in.

Thanks,
Jay

**Piping Error**
```
$ cat geom.csv > pipe
---------------------------------
$ tippecanoe -e tiles -zg -l points pipe
pipe:1: Found unexpected character
0 features, 10 bytes of geometry, 4 bytes of separate metadata, 0 bytes ofstring pool
Did not read any valid geometries
```

**Comment by ericfischer** [comment]
You're right, it currently uses the filename to determine the format. I don't think sniffing the format will be reliable, so probably the best thing I can do is add a `format` field to `--named-layer`.

**Comment by ericfischer** [comment]
This will be fixed in 3cb09d0 in https://github.com/mapbox/tippecanoe/pull/604

**Comment by jaykaron** [comment]
Thanks so much!

**Comment by mohesem** [comment]
Hello, 

does it work with PBF file as input ?

**Comment by ericfischer** [comment]
PBF (Geobuf) input can't be read from a stream, because the `vtzero` library works by mapping the entire structure into memory. If you are using Geobuf, you will have to provide it as a regular file.
